### PR TITLE
fix(linux): Fix Caps Lock

### LINKED ISSUE #4542: bug(linux): Caps Lock is not working as expected
This behavior is also experienced in Linux (`20.04.2 LTS`) 

_Originally posted by @MakaraSok in https://github.com/keymanapp/keyman/issues/4525#issuecomment-785773595_

### PR DESCRIPTION
**Pull request by ermshiperete**
Closes #4542.

### CODE REVIEW

File: linux/ibus-keyman/src/engine.c [hunk 1]
```diff
@@ -677,6 +677,10 @@ ibus_keyman_engine_process_key_event (IBusEngine     *engine,
             g_message("modstate KM_KBP_MODIFIER_LCTRL from lctrl_pressed");
         }
     }
+    if (state & IBUS_LOCK_MASK)
+    {
+        km_mod_state |= (1 << 8); // KM_KBP_MODIFIER_CAPS;
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply linux/ibus-keyman/src/engine.c:682]: Can we use the const rather than a hard coded value?

[the remark this answers is not in the document]
Comment (mcdurdin) [reply linux/ibus-keyman/src/engine.c:682]: Suggest basing on #5053 and then just use `KM_KBP_MODIFIER_CAPS` yeah?

[the remark this answers is not in the document]
Comment (darcywong00) [reply linux/ibus-keyman/src/engine.c:682]: @mcdurdin Does `KM_KBP_MODIFIER_NOCAPS` ever get checked?

[the remark this answers is not in the document]
Comment (mcdurdin) [reply linux/ibus-keyman/src/engine.c:682]: No, `KM_KBP_MODIFIER_NOCAPS` is not applicable for an incoming event -- only for a keyboard rule.

**Comment by darcywong00** [comment]
Does this fix mean we can enable kmx unit tests 032 and 033?
I notice the kmx files aren't checked in and are disabled in
https://github.com/keymanapp/keyman/blob/b614ee7c53b7fbe4afd731c3681e6422dea5795a/common/core/desktop/tests/unit/kmx/meson.build#L50-L51

**Comment by mcdurdin** [comment]
> Does this fix mean we can enable kmx unit tests 032 and 033?

No, these will be tracked by #5052.

**Review by darcywong00 — APPROVED** [review]
lgtm

File: linux/ibus-keyman/src/engine.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ermshiperete) [reply linux/ibus-keyman/src/engine.c]: > Can we use the const rather than a hard coded value?

I was thinking the same thing. Done.

**Comment by mcdurdin** [comment]
Jenkins build has failed; please investigate before we merge!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.45-alpha](https://keyman.com/downloads/releases/alpha/15.0.45)

**Pull request merged**